### DISCUSSION TITLE
Canvas follows location behaivor

### DISCUSSION THREAD

**Discussion by ImpulseDeer** [opengisch/QField#3311] [Ideas]
Hi thanks for a nice slimed app for exploring gis data with integrated support for external gps sources!

Trying it out now and had one expectation not aligning with reallity. I expected the "Canvas follows location" feature to update the  canvas position with _every_ new recived location and thus have the location marker staying fixed in the middle of the screen. The current behavior made me think that the feature didn't work to begin with as the canvas only changed on the first recived location after pressing the button and not any of the small pertubations after that (until I recorded this video when I walked a bit longer for what I was thinking would be a bug report).

https://user-images.githubusercontent.com/111496869/189143516-616674cb-a272-4580-81ae-52ba2ea5a8ed.mp4

My usecase with a external gps reciver would not suffer from the mapping jumping around due to noise which I could see be a reason for doing it as it is now. Anyone else would want the canvas to update on every location update?

It would also be nice if the "Canvas follows location" feature did not quit when zooming or touching the screen to reactivate it but only when pressing the "Locate button" again. Now I keep disabling it when I didn't intend on disabling it.

**Comment by dfwsbylwy** [comment]
real-time drawing may increase computation and power consumption. Perhaps designer can pre-draw a larger area of the display buffer than the screen, and then can pan the buffer map on the screen to save the calculation and power consumption of real-time drawing.
